include/h2o/dsr.h

```diff
@@ -35,24 +35,35 @@
  */
 typedef struct st_h2o_dsr_req_t {
     /**
-     * The transport protocol for which DSR is performed (e.g., quic-29). This value indicates how the packet should be protected,
-     * aside from the cipher-suite being used.
+     * HTTP version of front-end connection
      */
-    uint32_t quic_version;
-    /**
-     * The cipher-suite being used, as the ID registered to the TLS Cipher Suites registry.
-     */
-    uint16_t cipher;
-    /**
-     * address from which QUIC packets should be sent
-     */
-    quicly_address_t address;
+    int http_version;
+    struct {
```

Comment:
Did you mean union?

Maybe it is due to the parser not being able to determine the HTTP version prior to parsing the version-specific attributes. But I might argue that the internal of a particular function should not cause strange effects on the design of a global structure.

In the parser, maybe we can parse version-specific values into version-specific variables, then, after validating the set of variables, assign that of the specified version to the union of `h2o_dsr_req_t`.